bevy_renet/src/lib.rs

```diff
@@ -24,9 +24,9 @@ impl Plugin for RenetServerPlugin {
     fn build(&self, app: &mut App) {
         app.init_resource::<Events<ServerEvent>>();
 
-        app.configure_set(RenetSet::Server.run_if(resource_exists::<RenetServer>()));
+        app.configure_set(Update, RenetSet::Server.run_if(resource_exists::<RenetServer>()));
```

Comment:
You configure set for Update, but use it in `PreUpdate`. I think that's the issue.

Reply:
Do we need to configure this with all default sets?
Like:
```rust
app.configure_set(PreUpdate, RenetSet::Server.run_if(resource_exists::<RenetServer>()));
app.configure_set(Update, RenetSet::Server.run_if(resource_exists::<RenetServer>()));
app.configure_set(PostUpdate, RenetSet::Server.run_if(resource_exists::<RenetServer>()));
```

This set is used to simplify the `run_if(resource_exists::<RenetServer>()`

Reply:
I guess we hit this issue:
https://github.com/bevyengine/bevy/issues/9139

Reply:
It seems we shouldn't be using SystemSets, and should just use directly `run_if(resource_exists::<RenetServer>` / `run_if(resource_exists::<RenetClient>`

Reply:
I probably should remove the set enum too then.